skyvern/forge/sdk/workflow/models/block.py

```
@@ -3037,10 +3038,36 @@ def _detect_file_type_from_url(self, file_url: str) -> FileType:
+        with open(file_path, "rb") as f:
+            raw_data = f.read(sample_size)
+
+        result = from_bytes(raw_data)
+        best_match = result.best()
+        if best_match and best_match.encoding:
+            return best_match.encoding
+
+        for encoding in ["utf-8", "cp1252", "latin-1"]:
+            try:
+                raw_data.decode(encoding)
+                return encoding
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)